middle_end/flambda2/simplify/lifting/lifted_constant.ml

```diff
@@ -294,7 +294,8 @@ let apply_projection t proj =
       | Block_load { index } ->
         T.meet_block_field_simple typing_env ~min_name_mode:Name_mode.normal ty
           index
-      | Project_value_slot { project_from = _; value_slot } ->
+      | Project_value_slot { project_from = _; value_slot; kind = _ } ->
+        (* CR mshinwell: could use [kind]? *)
```

Comment:
It could be used either for a consistency check (that the kind of the result type is the expected one), or for adding subkind constraints. However, subkind constraints would better fit at the closure creation time, not on projection.